x-pack/plugins/stack_connectors/public/connector_types/cases_webhook/steps/auth.tsx

```diff
@@ -21,87 +21,111 @@ import {
   const [{ config, __internal__ }] = useFormData({
-    watch: ['config.hasAuth', '__internal__.hasHeaders'],
+    watch: ['config.hasAuth', 'config.authType', '__internal__.hasHeaders'],
   });
 
   const hasHeadersDefaultValue = !!getFieldDefaultValue<boolean | undefined>('config.headers');
 
-  const hasAuth = config == null ? true : config.hasAuth;
+  const authTypeDefaultValue =
+    getFieldDefaultValue('config.hasAuth') === false
+      ? null
+      : getFieldDefaultValue('config.authType') ?? CasesWebhookAuthType.Basic;
+  const authType = config == null ? CasesWebhookAuthType.Basic : config.authType;
   const hasHeaders = __internal__ != null ? __internal__.hasHeaders : false;
 
+  useEffect(() => setFieldValue('config.hasAuth', Boolean(authType)), [authType, setFieldValue]);
+
   return (
     <span data-test-subj="authStep" style={{ display: display ? 'block' : 'none' }}>
       <EuiFlexGroup>
         <EuiFlexItem>
           <EuiTitle size="xxs">
             <h4>{i18n.AUTH_TITLE}</h4>
           </EuiTitle>
-          <EuiSpacer size="m" />
-          <UseField
-            path="config.hasAuth"
-            component={Field}
-            config={{ defaultValue: true, type: FIELD_TYPES.TOGGLE }}
-            componentProps={{
-              euiFieldProps: {
-                label: i18n.HAS_AUTH,
-                disabled: readOnly,
-                'data-test-subj': 'hasAuthToggle',
-              },
-            }}
-          />
         </EuiFlexItem>
       </EuiFlexGroup>
-      {hasAuth ? (
-        <EuiFlexGroup justifyContent="spaceBetween">
-          <EuiFlexItem>
-            <UseField
-              path="secrets.user"
-              config={{
-                label: i18n.USERNAME,
-                validations: [
-                  {
-                    validator: emptyField(i18n.USERNAME_REQUIRED),
-                  },
-                ],
-              }}
-              component={Field}
-              componentProps={{
-                euiFieldProps: { readOnly, 'data-test-subj': 'webhookUserInput', fullWidth: true },
-              }}
-            />
-          </EuiFlexItem>
-          <EuiFlexItem>
-            <UseField
-              path="secrets.password"
-              config={{
-                label: i18n.PASSWORD,
-                validations: [
-                  {
-                    validator: emptyField(i18n.PASSWORD_REQUIRED),
-                  },
-                ],
-              }}
-              component={PasswordField}
-              componentProps={{
-                euiFieldProps: { readOnly, 'data-test-subj': 'webhookPasswordInput' },
-              }}
-            />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      ) : null}
+      <EuiSpacer size="m" />
+      {/* I wonder if this is necessary but I am leaving it for now because the Webhook connector does too.*/}
+      <UseField path="config.hasAuth" component={() => null} />
```

Comment:
I think this is there because the field is hidden (not shown in the UI). You can use `HiddenField` exported from `@kbn/es-ui-shared-plugin/static/forms/components`.

Reply:
Right but if the value is set on line 95 is it still required to be here?

Reply:
Sorry what I mean is  `<UseField path="config.hasAuth" component={HiddenField} />` instead of `<UseField path="config.hasAuth" component={() => null} />`